File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/internal/request/ReconcileRequest.java
package com.emerchantpay.gateway.genesisandroid.api.internal.request;


import com.emerchantpay.gateway.genesisandroid.api.util.Request;
import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

import java.util.List;
import java.util.Map;

public class ReconcileRequest extends Request {

	private String uniqueId;

	public ReconcileRequest() {
		super();
	}

	public ReconcileRequest setUniqueId(String uniqueId) {
		this.uniqueId = uniqueId;
		return this;
	}

	@Override
	public String getTransactionType() {
		return "wpf_reconcile";
	}

	@Override
	public String toXML() {
		return buildRequest("wpf_reconcile").toXML();
	}

	@Override
	public String toQueryString(String root) {
		return buildRequest(root).toQueryString();
	}

	protected RequestBuilder buildRequest(String root) {

		return new RequestBuilder(root).addElement("unique_id", uniqueId);
	}

	public List<Map.Entry<String, Object>> getElements() {
		return buildRequest("").getElements();
	}
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/internal/request/ReconcileRequest.kt
package com.emerchantpay.gateway.genesisandroid.api.internal.request


import com.emerchantpay.gateway.genesisandroid.api.util.Request
import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

class ReconcileRequest : Request() {

    private var uniqueId: String? = null

    val elements: MutableList<Map.Entry<String, Any>>?
        get() = buildRequest("")?.elements

    fun setUniqueId(uniqueId: String): ReconcileRequest {
        this.uniqueId = uniqueId
        return this
    }

    override fun toXML(): String? {
        return buildRequest("wpf_reconcile")?.toXML()
    }

    override fun toQueryString(root: String): String {
        return buildRequest(root)?.toQueryString().toString()
    }

    private fun buildRequest(root: String): RequestBuilder? {

        return uniqueId?.let { RequestBuilder(root).addElement("unique_id", it) }
    }

    override fun getTransactionType(): String? {
        return "wpf_reconcile"
    }
}
